group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -3100,6 +3100,15 @@ void validateDeleteGroup(String groupId) throws ApiException {
         group.validateDeleteGroup();
     }
 
+    public void cleanupGroupMetadata(List<Record> records) {
+        // TODO: committed offset?
+        groups.values().forEach(group -> {
+            if (group.eligibleForExpiration()) {
+                records.addAll(group.createGroupTombstoneRecords());
+            }
+        });
```

Comment:
It is a bit annoying that we have to iterate over all the groups to find the one eligible for expiration. Have you thought about alternatives?